common/json_param.c

```diff
@@ -439,6 +439,20 @@ struct command_result *param_string(struct command *cmd, const char *name,
 	return NULL;
 }
 
+struct command_result *param_invstring(struct command *cmd, const char *name,
+				       const char * buffer, const jsmntok_t *tok,
+				       const char **str)
+{
+	/* FIXME: there are a couple of things here
+	 * - we should really have in ccan a function to strip a prefix;
+	 * - I am fall in some check bunds error with tal so I can not
+	 * reuse the same *str */
+	const char *strtmp = tal_strndup(cmd, buffer + tok->start,
```

Comment:
json_strdup() is neater, and this leaks.  Allocate off NULL, then use take() below:

Reply:
Ah right, this is the bounds that I was breaking, mh I guess this is something that I did not understand of ccan, I need to scan again the code :)

Thanks for the mem check fix :)